api/runner/protocol/factory.go

```diff
@@ -0,0 +1,35 @@
+package protocol
+
+import (
+	"errors"
+	"io"
+
+	"github.com/iron-io/functions/api/models"
+)
+
+var errInvalidProtocol = errors.New("Invalid Protocol")
+
+// ContainerIO defines the interface used to talk to a hot container.
+// Internally, a protocol must know when to alternate between stdin and stdout.
+// It returns any protocol error, if present.
+type ContainerIO interface {
+	Dispatch(stdin io.Reader, stdout io.Writer) error
+}
+
+type Protocol string
+
+const (
+	HTTP Protocol = models.FormatHTTP
```

Comment:
It is. Check:
https://github.com/iron-io/functions/pull/332/files#diff-c68879dad563371d9aa7e11002087e18R99